src/ol/controls/RoutingControl.js

```diff
@@ -0,0 +1,419 @@
+import { Vector } from 'ol/layer';
+import { Circle, Stroke, Style, Fill } from 'ol/style';
+import { Vector as VectorSource } from 'ol/source';
+import { Feature } from 'ol';
+import { LineString, Point } from 'ol/geom';
+import { Modify } from 'ol/interaction';
+import { unByKey } from 'ol/Observable';
+import { click } from 'ol/events/condition';
+import { fromLonLat, toLonLat } from 'ol/proj';
+import lineSlice from '@turf/line-slice';
```

Comment:
I would try to use consistent naming for turf imports, something like...
```suggestion
import turfLineSlice from '@turf/line-slice';
```